apps/asap-server/src/migrations/1618488149-publish-date-to-added-date.ts

```diff
@@ -0,0 +1,48 @@
+/* istanbul ignore file */
+import { RestResearchOutput, Results, Squidex } from '@asap-hub/squidex';
+import { Migration } from '../handlers/webhooks/webhook-run-migrations';
+import logger from '../utils/logger';
+
+export default class MoveResearchOutputTextToDescription extends Migration {
+  up = async (): Promise<void> => {
+    const squidexClient = new Squidex<RestResearchOutput>('research-outputs', {
+      unpublished: true,
+    });
+
+    let pointer = 0;
+    let result: Results<RestResearchOutput>;
+
+    do {
+      result = await squidexClient.fetch({
+        $top: 10,
+        $skip: pointer,
+        $orderby: 'created asc',
+      });
+
+      for (const researchOutput of result.items) {
+        const publishDate = researchOutput.data.publishDate?.iv;
+        if (!publishDate) {
+          continue;
+        }
+
+        try {
+          const addedDate = publishDate.includes(' ')
+            ? new Date(`${publishDate} UTC`).toISOString() // for dates in format 'Month Year'
+            : new Date(publishDate).toISOString(); // For dates in ISO format
+
+          await squidexClient.patch(researchOutput.id, {
+            addedDate: { iv: addedDate },
+          });
+        } catch (err) {
+          logger.error(err, `Error migrating RO: ${researchOutput.id}`);
```

Comment:
it's a little risky because this way if errors are encountered rerunning the migration will require rollback (which actually isn't so bad either since the rollback doesn't do anything :D), but im just pointing it out

Reply:
This block is here just in case there's an error creating a date. (e.g. `publishDate = 'foo bar'`) In this case I don't want to stop the migration because of a bad record. Just log it and move on.
If the patch throws, then that's also fine. Nothing was changed in that case.